src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
```

Comment:
PullProofAI

The logging logic can be extracted into a separate method for better modularity and readability.
```diff
+   logRequestDetails(req: Request) {
+       // Logging logic here
+   }
+
+   use(req: Request, res: Response, next: NextFunction) {
+       const token = req.headers?.authorization?.split(' ')[1] || '';
+       
+       try {
+           const userDetail = this.jwt.decode(token) || {};
+           this.logRequestDetails(req, userDetail);
+       } catch (error) {
+           // Handle decoding error
+       }
+
+       next();
+   }
```


<!-- This is an auto-generated comment by PullProofAI -->